### PR TITLE
Add Month filter to events

### PR DESCRIPTION
**Pull request by bradleysigma**
image
Also arguments are now case insensitive (`!events all` equivalent to `!events All`).
Mild formatting improvements (line lengths, mostly).
Added `@loading_status`.

### CODE REVIEW

File: uqcsbot/scripts/events.py [hunk 1]
```diff
@@ -6,42 +6,55 @@
...
+CALENDAR_URL = ("https://calendar.google.com/calendar/ical/"
+                "q3n3pce86072n9knt3pt65fhio%40group.calendar.google.com/public/basic.ics")
+FILTER_REGEX = re.compile('full|all|[0-9]+( weeks?)?|jan.*|feb.*|mar.*|apr.*|may.*|jun.*'
+                          '|jul.*|aug.*|sep.*|oct.*|nov.*|dec.*')
 BRISBANE_TZ = timezone('Australia/Brisbane')
-MONTHS = ["JAN", "FEB", "MAR", "APR", "MAY", "JUN", "JUL", "AUG", "SEP", "OCT", "NOV", "DEC"]
+# empty string to one-index
+MONTH_ABBR = ["", "JAN", "FEB", "MAR", "APR", "MAY", "JUN",
+              "JUL", "AUG", "SEP", "OCT", "NOV", "DEC"]
+MONTH_FULL = ["", "January", "February", "March", "April", "May", "June",
```
Comment (nicklambourne) [inline uqcsbot/scripts/events.py]: calendar.month_name

File: uqcsbot/scripts/events.py [hunk 2]
```diff
@@ -6,42 +6,55 @@
...
-CALENDAR_URL = "https://calendar.google.com/calendar/ical/q3n3pce86072n9knt3pt65fhio%40group.calendar.google.com/public/basic.ics"  # noqa
-FILTER_REGEX = re.compile('(full|all|[0-9]+( weeks?)?)')
+CALENDAR_URL = ("https://calendar.google.com/calendar/ical/"
+                "q3n3pce86072n9knt3pt65fhio%40group.calendar.google.com/public/basic.ics")
+FILTER_REGEX = re.compile('full|all|[0-9]+( weeks?)?|jan.*|feb.*|mar.*|apr.*|may.*|jun.*'
+                          '|jul.*|aug.*|sep.*|oct.*|nov.*|dec.*')
 BRISBANE_TZ = timezone('Australia/Brisbane')
-MONTHS = ["JAN", "FEB", "MAR", "APR", "MAY", "JUN", "JUL", "AUG", "SEP", "OCT", "NOV", "DEC"]
+# empty string to one-index
+MONTH_ABBR = ["", "JAN", "FEB", "MAR", "APR", "MAY", "JUN",
```
Comment (nicklambourne) [inline uqcsbot/scripts/events.py]: calendar.month_abbr

File: uqcsbot/scripts/events.py [hunk 3]
```diff
@@ -6,42 +6,55 @@
...
+              "JUL", "AUG", "SEP", "OCT", "NOV", "DEC"]
+MONTH_FULL = ["", "January", "February", "March", "April", "May", "June",
+              "July", "August", "September", "October", "November", "December"]
+MONTH_NUMBER = {"jan": 1, "feb": 2, "mar": 3, "apr": 4, "may": 5, "jun": 6,
```
Comment (nicklambourne) [inline uqcsbot/scripts/events.py]: Can get from index of previous arrays, at worst, build it programatically:
`{month.lower():index for index, month in enumerate(MONTH_ABBR)}`

File: uqcsbot/scripts/events.py (hunk 1 above)
Comment (nicklambourne) [inline uqcsbot/scripts/events.py]: calendar.month_name

File: uqcsbot/scripts/events.py (hunk 2 above)
Comment (nicklambourne) [inline uqcsbot/scripts/events.py]: calendar.month_abbr

File: uqcsbot/scripts/events.py [hunk 4]
```diff
@@ -6,42 +6,55 @@
...
 from uqcsbot import bot, Command
-from uqcsbot.utils.command_utils import UsageSyntaxException
+from uqcsbot.utils.command_utils import UsageSyntaxException, loading_status
 from uqcsbot.utils.itee_seminar_utils import (get_seminars, HttpException, InvalidFormatException)
 
-CALENDAR_URL = "https://calendar.google.com/calendar/ical/q3n3pce86072n9knt3pt65fhio%40group.calendar.google.com/public/basic.ics"  # noqa
-FILTER_REGEX = re.compile('(full|all|[0-9]+( weeks?)?)')
+CALENDAR_URL = ("https://calendar.google.com/calendar/ical/"
+                "q3n3pce86072n9knt3pt65fhio%40group.calendar.google.com/public/basic.ics")
+FILTER_REGEX = re.compile('full|all|[0-9]+( weeks?)?|jan.*|feb.*|mar.*|apr.*|may.*|jun.*'
```
Comment (nicklambourne) [inline uqcsbot/scripts/events.py]: If you're going to be doing your own regex pattern matching, we're going to want to see some tests to confirm it works as intended.

File: uqcsbot/scripts/events.py (hunk 3 above)
Comment (bradleysigma) [reply uqcsbot/scripts/events.py]: Fixed.

File: uqcsbot/scripts/events.py (hunk 2 above)
Comment (bradleysigma) [reply uqcsbot/scripts/events.py]: Fixed.

File: uqcsbot/scripts/events.py (hunk 4 above)
Comment (bradleysigma) [reply uqcsbot/scripts/events.py]: Added tests.

**Pull request merged**